### PR TITLE
Restructure CMP0M code

### PR DESCRIPTION
**Pull request by sriharshakandala**
## Purpose 
Restructure `CloudMicrophysics 0M` code

----
- [x] I have read and checked the items on the review checklist.

### CODE REVIEW

File: src/parameters/cmp_0m.jl [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+struct CloudMicrophysicsParameters0M{FT} <: AbstractCloudMicrophysicsParameters
+    "precipitation timescale"
+    τ_precip::FT
+    "specific humidity precipitation threshold"
+    qc_0::FT
+    "supersaturation precipitation threshold"
+    S_0::FT
+end
+
+CloudMicrophysicsParameters0M(param_set) = CloudMicrophysicsParameters0M(
```
Comment (sriharshakandala) [inline src/parameters/cmp_0m.jl:18]: I think we should construct this directly from the `.toml` parameters file, but I would prefer to do that in a different PR!

**Comment by sriharshakandala** [comment]
bors try

File: src/parameters/cmp_0m.jl [hunk 2]
```diff
@@ -0,0 +1,22 @@
+"""
+    CloudMicrophysicsParameters0M{FT}
+
+Parameters for zero-moment bulk microphysics scheme
+
+# Fields
+$(DocStringExtensions.FIELDS)
+"""
+struct CloudMicrophysicsParameters0M{FT} <: AbstractCloudMicrophysicsParameters
+    "precipitation timescale"
```
Comment (trontrytel) [inline src/parameters/cmp_0m.jl:10]:
```suggestion
    "precipitation timescale [s]"
```

File: src/parameters/cmp_0m.jl [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    τ_precip::FT
+    "specific humidity precipitation threshold"
```
Comment (trontrytel) [inline src/parameters/cmp_0m.jl:12]:
```suggestion
    "specific humidity precipitation threshold [-]"
```

File: src/parameters/cmp_0m.jl [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    qc_0::FT
+    "supersaturation precipitation threshold"
```
Comment (trontrytel) [inline src/parameters/cmp_0m.jl:14]:
```suggestion
    "supersaturation precipitation threshold [-]"
```

**Review by trontrytel — APPROVED** [review]
